scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/authentication/ManageUsers.java

```diff
@@ -226,62 +273,152 @@ private static void askInteractively(Console console, UserInfo userInfo, Action
         try {
-            Properties props = new Properties();
-            try {
-                FileInputStream stream = new FileInputStream(new File(loginFilePath));
-                props.load(stream);
-                stream.close();
+            Properties destinationLoginProps = new Properties();
+            try (InputStreamReader stream = new InputStreamReader(new FileInputStream(loginFilePath))) {
+                destinationLoginProps.load(stream);
             } catch (Exception e) {
                 exitWithErrorMessage("could not read login file : " + loginFilePath, null, e);
             }
 
-            Multimap<String, String> groupsMap = loadGroups(groupFilePath);
+            Multimap<String, String> destinationGroupsMap = loadGroups(groupFilePath);
 
-            switch (action) {
-                case CREATE:
-                    if (props.containsKey(userInfo.getLogin())) {
-                        warnWithMessage("USER already exists in login file : " + loginFilePath + ", updating this user information.");
-                    }
-                    updateUserPassword(pubKey, userInfo.getLogin(), userInfo.getPassword(), props);
-                    System.out.println("Created user " + userInfo.getLogin() + " in " + loginFilePath);
-                    updateUserGroups(userInfo.getLogin(), userInfo.getGroups(), groupsMap);
-                    break;
-                case UPDATE:
-                    if (!props.containsKey(userInfo.getLogin())) {
-                        if (userInfo.getPassword() != null && userInfo.getGroups() != null) {
-                            warnWithMessage("USER does not exist in login file : " + loginFilePath + ", create this user.");
-                        } else {
-                            exitWithErrorMessage("USER does not exist in login file : " + loginFilePath + " and not enough information were provided to create a new user.", null, null);
-                        }
-                    }
-                    if (userInfo.getPassword() != null) {
-                        updateUserPassword(pubKey, userInfo.getLogin(), userInfo.getPassword(), props);
-                        System.out.println("Changed password for user " + userInfo.getLogin() + " in " + loginFilePath);
-                    }
-                    if (userInfo.getGroups() != null) {
-                        updateUserGroups(userInfo.getLogin(), userInfo.getGroups(), groupsMap);
-                    }
-                    break;
-                case DELETE:
-                    if (!props.containsKey(userInfo.getLogin())) {
-                        warnWithMessage("USER does not exist in login file : " + loginFilePath);
+            Properties sourceLoginProps = new Properties();
+            if (sourceLoginFile != null) {
+                try (InputStreamReader stream = new InputStreamReader(new FileInputStream(sourceLoginFile))) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this nested try block into a separate method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1141)